pkg/api/login.go

```diff
@@ -81,6 +81,13 @@ func (hs *HTTPServer) LoginView(c *models.ReqContext) {
 		return
 	}
 
+	urlParams := c.Req.URL.Query()
+	if _, isInternal := urlParams["internal"]; isInternal {
```

Comment:
I'm not sure about the `internal` name, not sure what to call it, but I'm thinking something like something like `noRedirect`?

Reply:
`noRedirect` sounds like we only disable redirect, but allow to login with options other than login/password. But in this case OAuth disabled on the login page, so user can only use login/password.

Reply:
Maybe something like `disableAutoLogin` or `skipAutoLogin`?

Reply:
> But in this case OAuth disabled on the login page, so user can only use login/password.

Ah, I didn't know that. In that case `internal` makes a little more sense to me.

`disableAutoLogin` sounds good to me.